MA-1113: add prop that hides time and only allows Date range

mihai-peteu opened the pull request:
# Summary
https://konghq.atlassian.net/browse/MA-1113

Adds the ability to only show the Custom (aka calendar) date range selection via the `mode` parameter. 
Previously, whenever we had both Relative and Custom tabs, we would default to showing a calendar _with_ time selection.

Also - we wanted to show the TZ at the end of the displayed value per https://konghq.atlassian.net/browse/MA-1009

### Example

The new `relativeDate` mode can be seen here:
https://deploy-preview-906--kongponents.netlify.app/components/datetime-picker.html#custom-date-and-relative-time-frames

<img width="695" alt="Screen Shot 2022-11-17 at 9 41 38 AM" src="https://user-images.githubusercontent.com/103061463/202518732-703f46ab-787b-4d6d-b727-eac511751b3e.png">

And the previous Date _and_ Time calendar selection has now become `mode="relativeDateTime"`:
https://deploy-preview-906--kongponents.netlify.app/components/datetime-picker.html#custom-date-time-and-relative-time-frames

## PR Checklist

* [ ] Does not introduce dependencies
* [ ] **Functional:** all changes do not break existing APIs and if so, bump major version.
* [ ] **Tests pass:** check the output of yarn test
* [ ] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
* [ ] **Framework style:** abides by the essential rules in Vue's style guide
* [ ] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs, debugger), or leftover comments
* [ ] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

adamdehaven commented on docs/components/datetime-picker.md:
```diff
@@ -276,7 +338,10 @@ A `Number` which sets the custom interval for the Minutes select dropdown.
 
-Required `String` prop which accepts the following values: `date`, `time`, `dateTime` and `relative`. The `relative` value is meant to denote an datetime picker instance which only contains relative time frames.
+Required `String` prop which accepts the following values: `date`, `time`, `dateTime`, `relative`. 
+- `relativeOnly`: denotes a datetime picker instance which only contains relative time frames
+- `relativeDate`: instance with relative time frames, and a calendar that supports date selection
+- `relativeDateTime`: instance with relative time frames, and a calendar that supports date and time selection
```
I don't understand these docs. The sentence above this lists 4 values that are valid, but then this list shows 3 additional values...?

mihai-peteu replied:
Sorry, I cleaned this up to be a list of all the allowed values. Less confusing now.

adamdehaven commented on src/components/KDateTimePicker/KDateTimePicker.vue:
```diff
@@ -227,14 +227,16 @@ export default defineComponent({
       required: true,
       validator: (value: string): boolean => {
-        return ['relative', 'date', 'time', 'dateTime'].includes(value)
+        return ['relativeOnly', 'relativeDateTime', 'relativeDate', 'date', 'time', 'dateTime'].includes(value)
```
What is `relativeOnly`? If it's just time, then maybe `relativeTime` instead?

mihai-peteu replied:
`relativeOnly` literally only shows the time frame buttons

I'll rename the other two to be `relativeAndCustomDate` | `relativeAndCustomDateTime` to be ultra specific :)

adamdehaven replied:
TBH that seems just as confusing, if not more. Can we be more concise somehow?

mihai-peteu replied:
settled on 

```
date
time
dateTime
relative
relativeDate
relativeDateTime
```

adamdehaven commented on src/components/KDateTimePicker/KDateTimePicker.vue:
```diff
@@ -482,11 +487,24 @@ export default defineComponent({
     const handleClose = async (): Promise<void> => {
-      state.hidePopover = true
+      // debugger
+      console.log('>>>>> handleClose')
+      // state.hidePopover = true
       await nextTick()
-      state.hidePopover = false
+      // state.hidePopover = false
```
nit: remove logging and other unneeded code?

mihai-peteu commented:
@adamdehaven addressed the comments and also added a new `date-fns-tz` dependency 😬  for this ticket: https://konghq.atlassian.net/browse/MA-1009

I can explain offline if need be.

mihai-peteu commented:
@adamdehaven - The confusing comment has been updated :)

adamdehaven commented on src/components/KDateTimePicker/KDateTimePicker.vue:
```diff
@@ -545,7 +554,7 @@ export default defineComponent({
 @import '@/styles/mixins';
 
-$timepicker-min-width: 360px;
+$timepicker-min-width: 400px;
```
non-blocking: could this have an impact on smaller screens or the component overflowing its parent?

mihai-peteu replied:
Reverting, i can always override min-width in khcp-ui.
